rbi/core/array.rbi

```diff
@@ -734,7 +734,11 @@ class Array < Object
   # You must not mix the two modes at a time; the block must always return
   # either true/false, or always return a number. It is undefined which value is
   # actually picked up at each iteration.
-  def bsearch; end
+  sig do
+    params(blk: T.proc.params(arg0: Elem).returns(T::Boolean))
```

Comment:
For this, and the other two signatures, returning a number should also be possible, so something like

```ruby
sig do
    params(blk: T.proc.params(arg0: Elem).returns(T.any(Float, Integer, T::Boolean)))
    .returns(T.nilable(Elem))
end
def bsearch(&blk); end
```

might be better

Reply:
works for me. the docs say:

```ruby
# You must not mix the two modes at a time; the block must always return
# either true/false, or always return a number
```

Reply:
Actually, now that I think about it again, having two signatures might be better given the documentation quoted by @jez:

```ruby
sig do
    params(blk: T.proc.params(arg0: Elem).returns(Numeric))
    .returns(T.nilable(Elem))
end
sig do
    params(blk: T.proc.params(arg0: Elem).returns(T::Boolean))
    .returns(T.nilable(Elem))
end
def bsearch(&blk); end
```